genai-perf/tests/test_llm_goodput_calculator.py

```diff
@@ -0,0 +1,150 @@
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from typing import Union
+
+import pytest
+from genai_perf.goodput_calculator.llm_goodput_calculator import LLMGoodputCalculator
+from genai_perf.metrics.llm_metrics import LLMMetrics
+
+
+def ns_to_sec(ns: int) -> Union[int, float]:
+    """Convert from nanosecond to second."""
+    return ns / 1e9
+
+
+class TestLLMGoodputCalculator:
+
+    TEST_METRIC = LLMMetrics(
+        request_throughputs=[10.12, 11.33],
+        request_latencies=[3, 44],
+        request_goodputs=[9.88, 10.23],
+    )
+
+    TEST_BENCHMARK_DURATION = 10
+
+    TEST_GOODPUT_CONSTRAINTS = {"request_latency": 10e-6}  # ms
+
+    def test_goodputcalculator_goodput(self) -> None:
+        """Test goodput property."""
+        gc = LLMGoodputCalculator(
+            goodput_constraints=self.TEST_GOODPUT_CONSTRAINTS,
+            metric=self.TEST_METRIC,
+            benchmark_duration=self.TEST_BENCHMARK_DURATION,
+        )
+
+        assert gc.goodput is None  # before computing
+        gc.compute()
+        assert gc.goodput == [0.1]  # after computing
+
+    def test_goodputcalculator_get_slo_base_name(self) -> None:
+        """Test get_slo_base_name method in GoodputCalculator class."""
+        gc = LLMGoodputCalculator(
+            goodput_constraints=self.TEST_GOODPUT_CONSTRAINTS,
+            metric=self.TEST_METRIC,
+            benchmark_duration=self.TEST_BENCHMARK_DURATION,
+        )
+
+        assert gc.get_slo_base_name("request_latency") == "request_latencies"
+        assert gc.get_slo_base_name("time_to_first_token") == "time_to_first_tokens"
+        assert gc.get_slo_base_name("inter_token_latency") == "inter_token_latencies"
+        assert gc.get_slo_base_name("output_token_throughput_per_request") == (
+            "output_token_throughputs_per_request"
+        )
+        with pytest.raises(KeyError):
+            gc.get_slo_base_name("hello1234")
+
+    def test_llmgoodputcalculator_compute(self) -> None:
+        """
+        Test compute method
```

Comment:
```suggestion
```
This comment can be trimmed down, since this line is redundant. This information is already in the function name, which is the ideal approach: self-documenting code.